cpp/src/arrow/flight/flight-test.cc

```diff
@@ -207,6 +207,51 @@ class TestFlightClient : public ::testing::Test {
   std::unique_ptr<TestServer> server_;
 };
 
+class AuthTestServer : public FlightServerBase {
+  Status DoAction(const ServerCallContext& context, const Action& action,
+                  std::unique_ptr<ResultStream>* result) {
+    std::vector<Result> results;
+    std::shared_ptr<Buffer> buf;
+    RETURN_NOT_OK(Buffer::FromString(context.peer_identity(), &buf));
+    results.push_back(Result{buf});
+    *result = std::unique_ptr<ResultStream>(new SimpleResultStream(std::move(results)));
+    return Status::OK();
+  }
+};
+
+class TestAuthHandler : public ::testing::Test {
+ public:
+  void SetUp() {
+    port_ = 30000;
+    server_.reset(new InProcessTestServer(
+        std::unique_ptr<FlightServerBase>(new FlightServerBase), port_));
+    ASSERT_OK(server_->Start(std::unique_ptr<ServerAuthHandler>(
+        new TestServerAuthHandler("user", "p4ssw0rd"))));
+    ASSERT_OK(ConnectClient());
+  }
+
+  void TearDown() { server_->Stop(); }
+
+  Status ConnectClient() { return FlightClient::Connect("localhost", port_, &client_); }
+
+ protected:
+  int port_;
+  std::unique_ptr<FlightClient> client_;
+  std::unique_ptr<InProcessTestServer> server_;
+};
+
+class TestAuth : public TestAuthHandler {
```

Comment:
Hmm... why are both `TestAuth` and `TestAuthHandler` needed here? `TestAuth` looks like it would work for all tests.